dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1200,11 +1200,102 @@ public boolean isEnabled() {
 
     public final static class TeePrintStream extends PrintStream {
         protected final TrOutputStream trStream;
+        protected final boolean autoFlush;
 
         public TeePrintStream(TrOutputStream trStream, boolean autoFlush) {
             super(trStream, autoFlush);
+            this.autoFlush = autoFlush;
             this.trStream = trStream;
         }
+
+        public void realFlush(String s) {
+            if (autoFlush) {
+                try {
+                    trStream.realFlush();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            if (autoFlush && (s.indexOf('\n') >= 0)) {
+                try {
+                    trStream.realFlush();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
```

Comment:
this method looks like it's flushing twice in the case of \n.  check for \n should probably be in the print methods (which decide whether or not to call realFlush), not in the realFlush method itself.